bin/warmup

```diff
@@ -0,0 +1,274 @@
+    if find_executable('bzip2') is None or find_executable('bunzip2') is None:
+        fatal('Please install bzip2 and bunzip2 to convert CSV files to Krun JSON format.')
+    if need_outliers:
+        pypy_path = find_executable('pypy')
+        if pypy_path is None:
+            warn('You do not appear to have PyPy installed. Some parts of this '
+                 'script may run slowly.')
+    if need_changepoints or need_plots:
+        try:
+            import numpy
+        except ImportError:
+            fatal('Please install the Python numpy library to generate changepoints and / or plots.')
+    if need_changepoints:
+        r_path = find_executable('R')
+        if r_path is None:
+            fatal('Please install R (e.g. r-base) to generate changepoints.')
+        try:
+            import rpy2
+        except ImportError:
+            fatal('Please install and the Python rpy2 library to generate changepoints.')
+    if need_latex:
+        pdflatex_path = find_executable('pdflatex')
+        if pdflatex_path is None:
+            fatal('You do not appear to have pdflatex installed, and so cannot '
+                  'compile LaTeX files. Please install LaTeX (e.g. tetex-live) and try again.')
+    if need_plots:
+        try:
+            import matplotlib
+        except ImportError:
+            fatal('Please install the Python matplotlib library to generate plots.')
+        try:
+            matplotlib.use('Agg')
+            import seaborn
+        except ImportError:
+            fatal('Please install the Python seaborn library to generate plots.')
+    return python_path, pypy_path, pdflatex_path, r_path
+
+
+class BenchmarkFile(object):
+    """Information about each benchmark, including CLI options to other scripts."""
+
+    def __init__(self, csv_filename, options, python_path, pypy_path, pdflatex_path, r_path):
+        self.basename = os.path.splitext(csv_filename)[0]
+        self.csv_filename = csv_filename
+        self.language = options.language
+        self.vm = options.vm
+        self.uname = options.uname
+        self.python_path = python_path
+        self.pypy_path = pypy_path
+        self.pdflatex_path = pdflatex_path
+        self.r_path = r_path
+
+    def check_input_file(self):
+        if not (os.path.isfile(self.csv_filename) and os.access(self.csv_filename, os.R_OK)):
+            fatal('File %s not found.' % self.csv_filename)
+
+    def convert_to_krun_json(self):
+        header, self.krun_filename = csv_to_krun_json([self.csv_filename],
+                                             self.language, self.vm, self.uname)
+        info('Writing out: %s' % self.krun_filename)
+        try:
+            self.iterations = int(header[-1]) + 1  # Iteration numbers start at 0.
+        except ValueError:
+            fatal('CSV file has malformed header. Run this script with --help for more details.')
+
+    def _get_output_filename(self, output):
```

Comment:
This is parsing the output of a script to find the output filename? This isn't ideal, but I could live with it.